Fix FederatedCredential() constructor options (#41016)
mdn/content: files/en-us/web/api/federatedcredential/federatedcredential/index.md

{{APIRef("Credential Management API")}}{{SeeCompatTable}}{{SecureContext_Header}}

The **`FederatedCredential()`** constructor creates a new {{domxref("FederatedCredential")}} object. In supporting browsers, an instance of this class may be passed the `credential` received from the `init` object for global {{domxref("Window/fetch", "fetch()")}}.

## Syntax

```js-nolint
new FederatedCredential(data)
```

### Parameters

- `data`

>   - : A {{domxref("FederatedCredentialInit")}} object. Options are:

Reviewer: We would usually say:

```suggestion
  - : A {{domxref("FederatedCredentialInit")}} object. An object with the following properties:
```

    - `name` {{optional_inline}}
      - : A string representing the credential username.
    - `iconURL` {{optional_inline}}
      - : A string representing the URL of an icon or avatar to be associated with the credential.
    - `origin`

>       - : A required string representing the credential's origin. {{domxref("FederatedCredential")}} objects are origin-bound, which means that they will only be usable on the specified origin they were intended to be used on.

Reviewer: We never note required-ness, it is the default on MDN.
Also is my other rewording here accurate? "only be usable on the specified origin they were intended to be used on" seemed weird to me. What actually is the origin they were intended to be used on?

```suggestion
      - : A string representing the credential's origin. {{domxref("FederatedCredential")}} objects are origin-bound, so they will only be usable on the origin specified here.
```

    - `provider`

>       - : A required string representing the credentials' federated identity provider (for example `"https://www.facebook.com"` or `"https://accounts.google.com"`).

Reviewer: From [the spec](https://w3c.github.io/webappsec-credential-management/#provider-identification) it looks like we can be more precise here?

> For consistency, federations passed into the APIs defined in this document (e.g. [FederatedCredentialRequestOptions](https://w3c.github.io/webappsec-credential-management/#dictdef-federatedcredentialrequestoptions)'s [providers](https://w3c.github.io/webappsec-credential-management/#dom-federatedcredentialrequestoptions-providers) array, or [FederatedCredential](https://w3c.github.io/webappsec-credential-management/#federatedcredential)'s [provider](https://w3c.github.io/webappsec-credential-management/#dom-federatedcredential-provider) property) MUST be identified by the [ASCII serialization](https://html.spec.whatwg.org/multipage/browsers.html#ascii-serialisation-of-an-origin) of the origin the provider uses for sign in. That is, Facebook would be represented by https://www.facebook.com and Google by https://accounts.google.com.

...so maybe we could say:

```suggestion
      - : A string which identifies the credentials' federated identity provider, specified as the origin that the provider uses to sign users in (for example `"https://www.facebook.com"` or `"https://accounts.google.com"`).
```

    - `protocol` {{optional_inline}}

>       - : A string representing the protocol of the credentials' federated identity provider (for example, `"openidconnect"`).

Reviewer: > (for example, `"openidconnect"`)

hmm. So can this be literally anything? It's not taken from a predefined list? I don't really understand how it can be usable for anything, if that is the case.

Reviewer: Yes, I couldn't see anything in the spec either.

What I mean is, can you pass "openid-connect" or "OpenID-Connect", and do they refer to the same protocol or not? It seems like they should but if the browser is expected to do anything with this option, it's not obvious how it could.

Anyway, not an issue we have to worry about in this PR.
